electrum/gui/qt/send_tab.py

```diff
@@ -319,6 +328,8 @@ def make_tx(fee_policy, *, confirmed_only=False, base_tx=None):
                 is_sweep=is_sweep,
                 send_change_to_lightning=self.config.WALLET_SEND_CHANGE_TO_LIGHTNING,
                 merge_duplicate_outputs=self.config.WALLET_MERGE_DUPLICATE_OUTPUTS,
+                password=password,
+                mind_silent_payments=True # always mind silent payments to detect sending to a previous sp-onchain addr
```

Comment:
Shouldn't `mind_silent_payments` be conditional on `any(o.is_silent_payment() for o in outputs)`?